bucketlistapi/resources.py

```diff
@@ -0,0 +1,323 @@
+from collections import OrderedDict
+from flask import g, request
+from flask_httpauth import HTTPTokenAuth
+from flask_restful import abort, Resource
+from sqlalchemy import and_
+from webargs.flaskparser import use_args
+
+
+from bucketlistapi.utils import (user_reg_login_field, name_field,
+                                 name_done_field, limit_field, save)
+from bucketlistapi.models import BucketList, BucketListItem, User
+from bucketlistapi import db
+
+auth = HTTPTokenAuth()
+
+
+@auth.verify_token
+def verify_token(token):
+    token = request.headers.get('Token')
+    if not token:
+        return False
+    user = User.verify_auth_token(token)
+    if not user:
+        return False
+    g.user = user
+    return True
+
+
+class UserRegAPI(Resource):
+    '''
+    The resource helps registers a new user with the post method.
+    [POST]
+
+    params: username and password.
+    '''
+
+    @use_args(user_reg_login_field)
+    def post(self, args):
+        # call methods to save user data and return success or otherwise
+        self.username = args.get('username')
+        self.password = args.get('password')
+        if self.check_user_exists():
+            abort(400, message='username already exists')
+        if (self.check_name() and
+                self.check_password()):
+            user = self.create_new_user()
+            return user.as_dict(), 201 if self.save_user(user) else abort(
+                400, message='bad request')
+        else:
+            abort(400, message='username/password minimum length is 4/8')
```

Comment:
This message `username/password minimum length is 4/8` is not enough to convey what is wrong to your users. Consider using some more descriptive message like `username was not supplied` or `You supplied an invalid username or password`. Users need to know the difference between supplying a field and supplying a badly-formatted field